framework/src/modules/chain/blocks/blocks.js

```diff
@@ -83,6 +84,7 @@ class Blocks extends EventEmitter {
 		this.genesisBlock = genesisBlock;
 		this.interfaceAdapters = interfaceAdapters;
 		this.slots = slots;
+		this.synchronizer = synchronizer;
```

Comment:
blocks should not reference synchronizer imo. Dependency should be opposite

Reply:
In that case we could fire an event instead of injecting the dependency.

Reply:
BTW I have no idea, how this change appears in this PR. As I didn't made it.